### PR TITLE
Generate default config if empty

### PR DESCRIPTION
**Pull request by g4s8**
#284 - Artipie checks if settings yaml configuration file is present, and generates default configuration settings if doesn't.

### CODE REVIEW

File: src/main/resources/example/repo/test/bin.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+repo:
+  type: file
+  storage: default
+  permissions:
+    "test":
```
Comment (olenagerasimova) [inline src/main/resources/example/repo/test/bin.yaml:5]: @g4s8 is it actually necessary to add quotes around `test`?

File: src/main/resources/example/repo/test/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+repo:
+  type: "docker"
+  storage: default
+  permissions:
+    \*:
+      - \*
```
Comment (olenagerasimova) [inline src/main/resources/example/repo/test/docker.yaml:6]: @g4s8 please replace \ with quotes, both asterisk should be written as "*"

File: src/main/resources/example/repo/test/maven.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+repo:
+  type: "maven"
+  storage: default
+  permissions:
+    \*:
+      - \*
```
Comment (olenagerasimova) [inline src/main/resources/example/repo/test/maven.yaml:6]: @g4s8 same here

File: src/main/java/com/artipie/VertxMain.java [hunk 4]
```diff
@@ -128,6 +129,36 @@ public static void main(final String... args) throws IOException, ParseException
         Logger.info(VertxMain.class, "Artipie was started on port %d", port);
     }
 
+    /**
+     * Find artipie settings.
+     * @param path Settings path
+     * @return Settings instance
+     * @throws IOException On read error
+     */
+    private static Settings settings(final Path path) throws IOException {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/VertxMain.java:138]: @g4s8 I'd suggest to extract this method into class and properly test it: now neither `JavaResource` no this method are tested

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) I see that you unassigned this issue; the order is still assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by g4s8** [comment]
@olenagerasimova thanks, I've fixed all your comments, but Github doesn't show my commits in this PR for some reasons: https://github.com/artipie/artipie/commit/5142fefc86f8066b2e927ec7d72599a0cda42dd3. I merged this branch to master manually because of this issue.

File: src/main/resources/example/repo/test/bin.yaml (hunk 1 above)
Comment (portlek) [reply src/main/resources/example/repo/test/bin.yaml:5]: No, it's not necessary.

Comment (portlek) [reply src/main/resources/example/repo/test/bin.yaml:5]: It's necessary when you type `#:->|$%&{}[]` in the key.

**Comment by 0crat** [comment]
Job was finished in 1 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)[/z](https://www.0crat.com/u/olenagerasimova) didn't make CR comments

**Comment by 0crat** [comment]
Pull request [#335](https://github.com/artipie/artipie/issues/335) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**